nibabel/nifti1.py

```diff
@@ -723,6 +723,11 @@ def set_data_shape(self, shape):
         If ``ndims == len(shape)`` then we set zooms for dimensions higher than
         ``ndims`` to 1.0
 
+        Nifti1 images can have up to seven dimensions. For Nifti surface files,
```

Comment:
So, there are Nifti1 surface files: http://nifti.nimh.nih.gov/pub/dist/src/niftilib/nifti1.h:

    #define NIFTI_INTENT_VECTOR    1007   /* for any other type of vector */

     /*! To signify that the vector value at each voxel is really a
         spatial coordinate (e.g., the vertices or nodes of a surface mesh):
           - dataset must have a 5th dimension
           - intent_code must be NIFTI_INTENT_POINTSET
           - dim[0] = 5
           - dim[1] = number of points
           - dim[2] = dim[3] = dim[4] = 1
           - dim[5] must be the dimensionality of space (e.g., 3 => 3D space).
           - intent_name may describe the object these points come from
             (e.g., "pial", "gray/white" , "EEG", "MEG").                   */

Annoyingly, this isn't how FreeSurfer actually implements its surfaces them, presumably because of the need for large vector hacks.